### DISCUSSION TITLE
How to build locust from git url

### DISCUSSION THREAD

**Discussion by tyge68** [locustio/locust#2959] [General]
Hi, I am trying to install directly from a git branch url or even just from a checkout.
However if I use the "pip install -e ." from the folder for example, once I start locust, I don't get the UI, just an "index.html" plain output on the main url.

Are there any specific commands to run before pip install -e . , in order to get the UI included ?

**Comment by cyberw** [comment]
See https://docs.locust.io/en/latest/developing-locust.html#install-locust-for-development

**Reply by tyge68** [reply]
Sure but that doesn't help , I want to create a docker image from my custom branch, in the past it was easy, now it seems to be more complex due to the new UI (maybe it also does more cross site scripting as I see some errors in the console)

**Reply by tyge68** [reply]
Ok it looks like now it need a lot of tooling pre-installed to build it
I opted for https://github.com/nikolaik/docker-python-nodejs docker image as the base that simplify the setup.

Still for an experience it's really a lot of dependencies and complex setup / tooling to build it. I miss the "one single" install command :)

**Reply by cyberw** [reply]
Yea, I'm not completely happy with it either. When we re-did the UI we moved from plain javascript to typescript and thus needed a build step. For a while we used the odd workaround of just commiting the built javascript, but that is also... horrible :)

I guess you've looked at locust own Dockerfile to see how we do it there? https://github.com/locustio/locust/blob/master/Dockerfile#L10

**Reply by tyge68** [reply]
Well I finally decided to simply build it outside with `poetry build` and copy the generated whl file in the docker image I need (using official docker python image as the base). Finally I usually deploy locust in multiple kubernetes clusters in many world regions so I can generate a small of load (150k RPS etc..)

**Reply by cyberw** [reply]
Cool stuff. If you’re interested, thats kind of what we’re doing over here at [locust.cloud](https://locust.cloud)

(we’re not multi region yet, but we can deifinitely churn out >100k rps :)
